src/recoil/sugi.ts

```diff
@@ -0,0 +1,89 @@
+import { atom, selector } from "recoil";
+import { dagStratify, sugiyama } from "d3-dag/dist";
+import { DagStateNodesLinksStrat } from "../components/OntologyExplorer/types";
+
+interface LayoutState {
+  nodeRadius: number;
+  layeringChoice: string;
+  decrossingsChoice: string;
+  coordsChoice: string;
+  scaleMultiplier: number;
+  sugiyamaWidthAspectRatio: number | null;
+  sugiyamaHeightAspectRatio: number | null;
+}
+
+export const dagDataStructureState = atom<DagStateNodesLinksStrat | null>({
+  key: "dagDataStructure",
+  default: null,
+});
+
+export const sugiyamaIsEnabledState = selector<boolean>({
+  key: "sugiyamaIsEnabled",
+  get: ({ get }) => {
+    const dagDataStructure = get(dagDataStructureState);
+    const sugiyamaRenderThreshold = get(sugiyamaRenderThresholdState);
+
+    return (
+      dagDataStructure?.sugiyamaStratifyData && dagDataStructure?.sugiyamaStratifyData.length < sugiyamaRenderThreshold
+    );
+  },
+});
+
+export const sugiyamaRenderThresholdState = atom<number>({
```

Comment:
Could you add a line explaining what is being thresholded here? I couldn't grok it. Is it the number of nodes?